### ISSUE TITLE
Project setup - "Invariant Violation" error when navigating to url after "yarn start:dev"

### ISSUE THREAD

**Issue by graftan** [project-koku/koku-ui#1676]
Hello all,

This project caught my eye and looks very interesting.  I was hoping to set it up on a dev vm and try it out with my aws account, however, I am running issues getting this ui project up and running.  

I am unsure if there are missing steps from the readme or if I am just being silly - details are below. If the readme needs to be updated or there is any other way I can contribute/help are you open for PRs?

Details:
* setup is all on a centos 8 vm
* koku api/server setup via Dockerfile and all containers up running
* insights proxy - but I am unsure it's exact purpose and how it ties in but it is running as per the instructions in the readme
* currently setting up this project koku-ui, after running "yarn start:dev", when I navigate to the page I get the following error 

> Uncaught Invariant Violation: Target container is not a DOM element.
>     React 2
>     <anonymous> webpack:///./src/index.tsx?:30
>     tsx http://ci.cloud.redhat.com:8002/beta/apps/cost-management/main.bundle.js:3951
>     __webpack_require__ http://ci.cloud.redhat.com:8002/beta/apps/cost-management/main.bundle.js:85
>     <anonymous> webpack:///multi_(webpack)-dev-server/client?:10
>     1 http://ci.cloud.redhat.com:8002/beta/apps/cost-management/main.bundle.js:5581
>     __webpack_require__ http://ci.cloud.redhat.com:8002/beta/apps/cost-management/main.bundle.js:85
>     checkDeferredModules http://ci.cloud.redhat.com:8002/beta/apps/cost-management/main.bundle.js:46
>     <anonymous> http://ci.cloud.redhat.com:8002/beta/apps/cost-management/main.bundle.js:222
>     <anonymous> http://ci.cloud.redhat.com:8002/beta/apps/cost-management/main.bundle.js:225
> ven

Further details:
* Node version = v12.18.3
* Yarn version = 1.22.5

Let me know if you need any further details or if I am being silly please kindly point me in the right direction.  Thanks in advance for any assistance you can provide.

**Comment by boaz0** [comment]
Hey @graftan I am afraid you will have to set the backend as well that includes at least koku and insight services.
I personally use an internal environment to just manually run my UI and see how it looks.

I will **try** and allocate time to see how one can run the UI on a local environment but I cannot promise anything

**Comment by graftan** [comment]
@boaz0 - thanks, appreciated.

Just to clarify, I am not trying to run koku-ui standalone.  On the same server I have setup the koku backend and it successfully runs e.g. http://localhost:8000/api/cost-management/v1/sources/ returns the relevant django api page response.  I also have the insight services proxy running as per the instructions in the repo README. 

Are referring to another backend component other than those that is needed to get koku-ui up and running?

**Comment by boaz0** [comment]
Yes, I believe you'll need this too https://github.com/RedHatInsights/insights-chrome

**Comment by boaz0** [comment]
@graftan - I'm closing this but if you still need help, feel free to re-open.

In addition, we are no longer following issues through Github and so if you find issues/problems/bugs please open them through our JIRA: https://issues.redhat.com/projects/COST/

thanks

**Comment by graftan** [comment]
@boaz0 thanks for the help,  the insights-chrome repo allowed me to get koku-ui running.  

I have another issue with regards to adding AWS sources - I have followed the guide:
https://koku.readthedocs.io/en/latest/sources.html#creating-an-iam-policy-and-role-for-cost-usage-consumption

The guide specifies aws account id 589173575009 - is this a Koku/Redhat account that perhaps has some sort of processing functionality needed for Koku?  
I run the aws-source command with AWS_RESOURCE_NAME variable set to the ARN I created in aws but get an error -  "Unable to access account resources with ARN..".  

I will create a ticket on Jira and link to this issue since this is now closed.  Once again, thanks for the help you provided - it is all much appreciated.

Hopefully once I'm setup and understand all the different components I can perhaps submit a PR including insights-chrome and any other details that may help others like me set it up a bit more easily.  I'm just not confident in doing so right now as I haven't grokked it all.

Cheers.
